jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -739,7 +739,7 @@ public MethodHandle findVirtual(Class<?> clazz, String methodName, MethodType ty
 					handle = new DirectHandle(clazz, methodName, type, MethodHandle.KIND_VIRTUAL, clazz, true);
 					int handleModifiers = handle.getModifiers();
 					/* Static check is performed by native code */
-					if (!Modifier.isPrivate(handleModifiers) && !Modifier.isFinal(handleModifiers)) {
+					if (!Modifier.isPrivate(handleModifiers) && !Modifier.isFinal(handleModifiers) && !Modifier.isFinal(clazz.getModifiers())) {
```

Comment:
This is subtle.  I missed the distinction between the `clazz.getModifiers()` & `handleModifiers`.  Can this be refactored to be more clear?

```java
handle = new DirectHandle(clazz, methodName, type, MethodHandle.KIND_VIRTUAL, clazz, true);
/* If the class is final, then there are no subclasses and the DirectHandle is sufficient */
if (!Modifier.isFinal(clazz.getModifiers()) {
    int handleModifiers = handle.getModifiers();
    /* Static check is performed by native code */
    if (!Modifier.isPrivate(handleModifiers) && !Modifier.isFinal(handleModifiers)) {
        ......
    }
}
```